src/Native/System.Native/CMakeLists.txt

```diff
@@ -18,9 +18,10 @@ add_library(System.Native
 
 if (CMAKE_SYSTEM_NAME STREQUAL Linux)
     target_link_libraries(System.Native rt)
-
     add_definitions(-DPLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS)
     add_definitions(-DPLATFORM_SUPPORTS_DUAL_MODE_IPV4_PACKET_INFO)
+elseif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    add_definitions(-DHAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR)
```

Comment:
This should be 0 or 1 like the other HAVE_* config macros. I don't want a mix of #if and #ifdef HAVE_* because it's easy to get wrong if it's not always the same pattern.

PLATFORM_SUPORTS_* should also probably be named to match.

I'd also prefer if all of these were handled in configure.cmake or pal_config.h.in.